### ISSUE TITLE
no scheduled run

### ISSUE THREAD

**Issue by henningkessler** [lindegroup/autopkgr#322]
hello,

I installed a AutoPkgr (version 1.2.1) onto an existing Autopkg installation (0.4.2) on an 10.9.5 OS X Server. Everything seams to run nicely. Only issue is that Autopkgr doesn't run the configured automatically updates. The LaunchDeamons are installed and loaded and when i run the command inside of the application or from the cli with 
```
/Applications/AutoPkgr.app/Contents/MacOS/AutoPkgr -runInBackground YES
```
everything seams to be okay.

the only thing I found irritating are the error messages in the console when I start the application.
```
 30.03.15 22:19:23,200 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77ac06ad0> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:23,202 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77af14960> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:23,203 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77ad07720> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:23,205 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77af16140> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:23,211 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77af19f10> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key OnLoad.
30.03.15 22:19:23,244 AutoPkgr[7004]: Invalid color System, labelColor (warning given only once)
30.03.15 22:19:24,543 WindowServer[138]: disable_update_timeout: UI updates were forcibly disabled by application "AutoPkgr" for over 1.00 seconds. Server has re-enabled them.
30.03.15 22:19:25,004 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77e206460> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:25,005 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77e208870> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:25,006 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77e209d90> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:25,008 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77e20aff0> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key cccTaskDict.
30.03.15 22:19:25,013 AutoPkgr[7004]: Exception Raised: [<AHLaunchJob 0x7fa77ac168a0> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key OnLoad.
30.03.15 22:19:25,043 WindowServer[138]: common_reenable_update: UI updates were finally reenabled by application "AutoPkgr" after 1.50 seconds (server forcibly re-enabled them after 1.00 seconds)
30.03.15 22:19:25,101 com.lindegroup.AutoPkgr.helper[7007]: xpc __securityd_create_connection_block_invoke got event: Connection invalid
30.03.15 22:19:25,105 com.lindegroup.AutoPkgr.helper[7007]:  SecOSStatusWith error:[-25291] The operation couldn’t be completed. (com.apple.security.xpc error 3 - <connection: 0x7f9d12418b50> { name = com.apple.securityd.xpc, listener = false, pid = 0, euid = 4294967295, egid = 4294967295, asid = 4294967295 }: Connection invalid)
30.03.15 22:19:25,162 secd[573]:  securityd_xpc_dictionary_handler AutoPkgr[7004] copy_matching The operation couldn’t be completed. (OSStatus error -34018 - client has neither application-identifier nor keychain-access-groups entitlements)
30.03.15 22:19:25,164 AutoPkgr[7004]:  SecOSStatusWith error:[-34018] The operation couldn’t be completed. (OSStatus error -34018 - Remote error : The operation couldn‚Äôt be completed. (OSStatus error -34018 - client has neither application-identifier nor keychain-access-groups entitlements))
```

**Comment by eahrold** [comment]
@henningkessler, 
Thanks for the report.

The AHLaunchJob exception is actually a caught exception an won't cause a problem. It's there because CarbonCopyCloner is using a value in it's launchd.plist that is not a valid key (cccTaskDict). In the next release it will get silenced. It's similar to this issue #319 

The  `SecOSStatusWith error:[-34018]` is a known Apple issue, and we've tried a number of things to silence it, but none have worked, so that will stick around. It's a duplicate of this issue #174.

We appreciate the info, and if you find anything else like this don't hesitate to post.

**Comment by henningkessler** [comment]
you are welcome and thanks for the explanations.

but do you have an idea why the automatically runs are not started?

**Comment by eahrold** [comment]
When you do `sudo launchctl list com.lindegroup.AutoPkgr.schedule` what's the `LastExitStatus` key?

**Comment by eahrold** [comment]
There is another possible explanation I just thought of. Is there a chance you enabled the schedule and then moved the AutoPkgr app to a different folder?

**Comment by henningkessler** [comment]
Hi eahrold,

thanks for all the hints. Even when this gets embarrassing  for me I have to say it just started to work yesterday evening after I finished my work. Is it possible that the scheduled runs don't start as long the application is active? Or maybe it started after I changed the schedule a couple of times? 

anyway this is the current (working) output:
```
[someone@somemachine:~]1001$ sudo launchctl list com.lindegroup.AutoPkgr.schedule
Password:
{
	"Label" = "com.lindegroup.AutoPkgr.schedule";
	"LimitLoadToSessionType" = "System";
	"OnDemand" = true;
	"LastExitStatus" = 0;
	"TimeOut" = 30;
	"Program" = "/Applications/AutoPkgr.app/Contents/MacOS/AutoPkgr";
	"ProgramArguments" = (
		"/Applications/AutoPkgr.app/Contents/MacOS/AutoPkgr";
		"-runInBackground";
		"YES";
	);
	"SessionCreate" = true;
};
```
and no I haven't moved the app to a different folder.

But I have to say thank you guys for all the work you are doing. Great help!
